README.md
# Cursix Tech
Simple AI bashing mod for Factorio

Factorio expects mod to be in a folder called cursix-tech_*version* and either placed in the mod directory as is or zipped up using the same naming scheme.

## Features:
- New Buildings
  - Cursix Beacon
    - 1x1 Beacon that consumes far less power and greater transmission range
- New Modules
  - Cursix Power Mushroom
    - Increases speed and reduces energy and pollution
  - Cursix 1-Up Mushroom
    - Doubles productivity
      - Not usable in normal beacons
  - Cursix Poison Mushroom
    - Slows speed, drastically increases pollution
- New Armor
  - Cursix Armor
    - Increased Inventory Space (30 more than Power Armor Mk2)
    - Massive resistance to damage
    - Massive Equipment Grid (12x12)
- New Personal Equipment
  - Sonic Scrambled Eggs
    - Massive personal power generator
  - Barrier Jacket
    - Massive personal shield generator
  - Energy Storage Crystal
    - Massive personal battery
  - Sonic Speed Shoes
    - Drastically smaller (1x1) personal speed booster
  - Cursix Axel Shooter
    - Improved Personal Defense System
  - Personal Moogle Roboport
    - Personal Roboport with greater range, drone capacity as well as faster recharge
  - Akane Eggs
    - Advance version of Sonic Scrambled Eggs
  -  Barrier Jacket Mk. 2
    - More powerful version of the Barrior Jacket
- New Research
  - Worker Robot Battery Research
    - Increases how long logistic and worker robots can stay out in one charge.
  - Cursix Crafting Bear
    - Increases player's crafting speed.

## Disclaimers
The graphics/icons/images in this mod were not made by the author and are copyrighted to their respected owners.

## Credits
- Cursix Beacon
  - Beacon model imported from [Logistic Technology Beacon](https://mods.factorio.com/mod/logtech-beacon) mod.
- Cursix Power Mushroom, 1-Up Mushroom and Poison Mushroom
  - Super Mario Brothers from Nintendo
- Barrier Jacket, Barrier Jacket Mk 2, Cursix Axel Shooter
  - Magical Girl Lyrical Nanoha Series
- Sonic Speed Shoes
  - Sonic The Hedgehog from Sega
- Personal Moogle Roboport
  - Moogles taking from Final Fantasy from Square Enix

changelog.txt
---------------------------------------------------------------------------------------------------
Version: 1.0.18
Date: 
  Changes:
    - Removed SE Space Attribute from Cursix Beacon as it didn't work
---------------------------------------------------------------------------------------------------
Version: 1.0.17
Date: 02. 01. 2023
  Minor Features:
    - Cursix Beacon now compatible with Space Exploration
---------------------------------------------------------------------------------------------------
Version: 1.0.16
Date: 04. 12. 2022
  Changes:
    - Reverted Description change for Poison Mushroom
---------------------------------------------------------------------------------------------------
Version: 1.0.15
Date: 05. 03. 2022
  Changes:
    - Added Plasma Damage Type to Mod
    - Cursix Armor resistance adjusted
      - Immune to Electric
      - 1000/80% to Plasma
    - Description change for Poison Mushroom
---------------------------------------------------------------------------------------------------
Version: 1.0.14
Date: 10. 08. 2021
  Changes:
    - Cursix Equipment now compatible with Bob's Vehicle Equipment Mod
---------------------------------------------------------------------------------------------------
Version: 1.0.13
Date: 03. 08. 2021
  Changes:
    - Personal Moogle Roboport now supports up to 100 robots
      - Number of charging stations remain 50
---------------------------------------------------------------------------------------------------
Version: 1.0.12
Date: 31. 07. 2021
  Minor Features:
    - Cursix Tech Lvl 2 Added
      - Barrier Jacket Mk 2
      - Akane Eggs
    - Moogle Roboport
  Changes:
    - Cursix Armor no longer has durability
    - Some Description Changes
---------------------------------------------------------------------------------------------------
Version: 1.0.11
Date: 10. 07. 2021
  Minor Features:
    - Added Cursix Beacon
  Changes:
    - Barrier Jacket recipe changed
      - 10x Wood
      - 50x Eletronic Circuits
      - 1x Cursix Energy Storage Cyrstal
      - Craftime (at 1x): 25 seconds
    - Cursix Speed Shoes recipe changed
      - 2x Wood
      - 50x Iron Gears
      - 10x Coal
    - Cursix Armor recipe changed
      - 100x Iron Plates
      - 100x Copper Plates
      - 25x Wood
---------------------------------------------------------------------------------------------------
Version: 1.0.10
Date: 06. 07. 2021
  Minor Features:
    - Added Axel Shooter Equipment
  Changes:
    - Storage Crystal now stores 100GJ of power, charges/discharges at 5GW
---------------------------------------------------------------------------------------------------
Version: 1.0.9
Date: 06. 02. 2021
  Bugfixes:
    - Factorio 1.1 compatibility
---------------------------------------------------------------------------------------------------
Version: 1.0.8
Date: 14. 08. 2020
  Bugfixes:
    - Factorio 1.00 compatibility
---------------------------------------------------------------------------------------------------
Version: 1.0.7
Date: 10. 10. 2019
  Minor Features:
    - Added Cursix Crafting Bear Tech; increases character's crafting speed
  Bugfixes:
    - Worker Robot Battery 3 was avaible before Space Science Pack was researched
---------------------------------------------------------------------------------------------------
Version: 1.0.6
Date: 30. 09. 2019
  Changes:
    - 1up Mushroom now requires 10 stones instead of 50
    - Removed modifier-description.worker-robot-battery locale as base game now has it
---------------------------------------------------------------------------------------------------
Version: 1.0.5
Date: 29. 09. 2019
  Changes:
    - Barrier Jacket recipe changed from 10x Military Science Pack to 1x Automation and 1x Logistic Science Pack
    - Reduced gears needed for Sonic Shoes to 50
  Bugfixes:
    - Cursix Armor sort order was incorrect
---------------------------------------------------------------------------------------------------
Version: 1.0.4
Date: 27. 09. 2019
  Changes:
    - Cursix Armor and Cursix Mushroom recipes now use Automation Science Pack
---------------------------------------------------------------------------------------------------
Version: 1.0.3
Date: 25. 09. 2019
  Changes:
    - Corrected Cursix Armor Animation script for Factorio 0.17.x
    - Removed flags 'goes-to-main-inventory' from prototypes
    - Fixed Energy Storage Cyrstal electric usage priority
    - Removed raw-wood and replaced with wood for Recipes
    - Science Packs for Recipes and Technology Fixed
---------------------------------------------------------------------------------------------------
Version: 1.0.2
Date: 19. 03. 2018
  Minor Features:
    - Added Worker Robot Battery Research.
---------------------------------------------------------------------------------------------------
Version: 1.0.1
Date: 13. 03. 2018
  Changes:
    - Internal name changes to various items for consistency.
        "sonic-scrambled-eggs-equipment" -> "cursix-scrambled-eggs-equipment"
        "barrier-jacket-shield-equipment" -> "cursix-barrier-jacket-equipment"
        "energy-storage-crystal-equipment" -> "cursix-energy-storage-crystal-equipment"
        "sonic-speed-shoes-equipment" -> "cursix-speed-shoes-equipment"
    - Updated Cursix Power Mushroom description.
    - Renamed "Barrier Jacket Shield" to "Barrier jacket".
    - Changed stack size of Barrier Jacket from 50 to 500.
    - Changed stack size of Sonic Speed Shoes from 200 to 500.
    - Changed stack size of Sonic Scrambled Eggs from 200 to 500.
    - Changed stack size of Energy Storage Crystal from 100 to 500.
    - Changes to Barrier Jacket:
        Max Shield: 25000 -> 50000
        Input Flow Limit: 555MW -> 600MW
        Buffer Capacity: 500MJ -> 2GJ
    - Energy Storage Crystal:
        Buffer Capacity: 10GJ -> 20GJ
---------------------------------------------------------------------------------------------------
Version: 1.0.0
Date: 03. 02. 2018

prototypes/entity/cursix-entities.lua
local sounds = require '__base__/prototypes/entity/sounds.lua'

data:extend(
{
  {
    type = "beacon",
    name = "cursix-beacon",
    icon = "__cursix-tech__/graphics/icons/cursix-beacon.png",
	  icon_size = 32,
    flags = {"placeable-player", "player-creation"},
    minable = {hardness = 0.2, mining_time = 0.5, result = "cursix-beacon"},
    max_health = 300,
    corpse = "small-remnants",
	  collision_box = {{-0.1, -0.1}, {0.1, 0.1}}, 
    selection_box = {{-0.5, -0.5}, {0.5, 0.5}},
	  drawing_box = {{-1.0, -2.0}, {1.0, -0.5}},
    resistances =
    {
      {
        type = "fire",
        percent = 80
      },
      {
        type = "impact",
        percent = 40
      }
    },
    allowed_effects = {"speed", "productivity", "consumption", "pollution"},
    
    base_picture = {
    	layers = {
        {
          filename = "__cursix-tech__/graphics/entity/cursix-beacon.png",
          priority = "extra-high",
          width = 30,
          height = 89,
          shift = util.by_pixel(-2, -39.5),
          hr_version = {
            filename = "__cursix-tech__/graphics/entity/hr-cursix-beacon.png",
            priority = "extra-high",
            width = 59,
            height = 178,
            shift = util.by_pixel(-2.25, -39.5),
            scale = 0.5,
          }
        },
        {
          filename = "__cursix-tech__/graphics/entity/cursix-beacon-shadow.png",
          priority = "extra-high",
          width = 119,
          height = 25,
          shift = util.by_pixel(52.5, -2.5),
          draw_as_shadow = true,
          hr_version = {
            filename = "__cursix-tech__/graphics/entity/hr-cursix-beacon-shadow.png",
            priority = "extra-high",
            width = 237,
            height = 50,
            shift = util.by_pixel(52.75, -3),
            draw_as_shadow = true,
            scale = 0.5,
          }
        }
      }
	  },
    
    radius_visualisation_picture =
    {
      filename = "__base__/graphics/entity/beacon/beacon-radius-visualization.png",
      priority = "extra-high-no-scale",
      width = 10,
      height = 10
    },
    
    supply_area_distance = 8,
    energy_source =
    {
      type = "electric",
      usage_priority = "secondary-input"
    },
    vehicle_impact_sound = sounds.generic_impact,
    open_sound = sounds.machine_open,
    close_sound = sounds.machine_close,
    working_sound =
    {
      sound =
      {
        {
          filename = "__base__/sound/beacon-1.ogg",
          volume = 0.2
        },
        {
          filename = "__base__/sound/beacon-2.ogg",
          volume = 0.2
        }
      },
      audible_distance_modifier = 0.33,
      max_sounds_per_type = 3
      -- fade_in_ticks = 4,
      -- fade_out_ticks = 60
    },
    energy_usage = "50kW",
    distribution_effectivity = 1.0,
    module_specification =
    {
      module_slots = 2,
      module_info_icon_shift = {0, 0},
      module_info_multi_row_initial_height_modifier = 1.0,
      module_info_max_icons_per_row = 2
    }
  }
})